### DISCUSSION TITLE
Unreasonable res0Norm

### DISCUSSION THREAD

**Discussion by nkulkar2** [Nek5000/nekRS#457] [Q&A]
Hello,

I am trying to setup a turbulent case and solving for the temperature equation. But I am facing unreasonable res0Norm error. The case works when I remove temperature from the *.par file. My *.par file and the useric and userbc are shown below:

[GENERAL] 
stopAt = endTime
#numStep = 0
endTime = 200.0
dt = 1e-03
variableDt = yes
timeStepper = bdf2
writeControl=runTime
writeInterval=5.0

regularization = hpfrt + nModes=1 + scalingCoeff=10

polynomialOrder = 7
#startFrom = channel0.f00003

[PRESSURE]
residualTol = 1e-05
residualProj = no

[VELOCITY]
residualTol = 1e-07
density = 1.0
viscosity = -10000
residualProj = no

[TEMPERATURE]
#ConjugateHeatTransfer = no
##advection = yes
conductivity = -8590.0 
rhoCp = 1.0
residualTol = 1.0e-6

c-----------------------------------------------------------------------
      subroutine userbc(ix,iy,iz,iside,ieg) ! set up boundary conditions
c     NOTE ::: This subroutine MAY NOT be called by every process

      implicit none

      integer ix,iy,iz,iside,ieg

      include 'SIZE'
      include 'TOTAL'
      include 'NEKUSE'

      real uin, vin, win
      integer e
      common /cvelbc/ uin(lx1,ly1,lz1,lelv)
     $              , vin(lx1,ly1,lz1,lelv)
     $              , win(lx1,ly1,lz1,lelv)

	  if (ifield.eq.1) then ! velocity field
        e = gllel(ieg)
        ux=uin(ix,iy,iz,e)
        uy=vin(ix,iy,iz,e)
        uz=win(ix,iy,iz,e)
	  elseif (ifield.eq.2) then ! temperature field
		 temp=1.0
      else
        write(6,*) 'ifield index is larger than the number of '
        write(6,*) 'fields set to 1 '
        call exitt()
      endif
      return
      end
c-----------------------------------------------------------------------
      SUBROUTINE USERIC (IX,IY,IZ,ieg)

C Set user-defined initial conditions

      implicit none

      integer ix,iy,iz,ieg

      INCLUDE 'SIZE'
      INCLUDE 'TOTAL'
      INCLUDE 'NEKUSE'
	  
	  real bpara,apara,kx,kz,eps

      eps = 1e-2
      kx  = 23
      kz  = 13

      apara = kx * 2*PI/10
      bpara  = kz * 2*PI/10

      if (ifield.eq.1) then ! velocity field
		  ux=0.0 + eps*bpara  * sin(apara*x)*cos(bpara*z)
		  uy=1.0 + eps*bpara  * sin(apara*x)*cos(bpara*z)
		  uz=0.0 - eps*bpara  * sin(apara*x)*cos(bpara*z)
      elseif (ifield.eq.2) then ! temperature field
		  temp=1.0
      else
        write(6,*) 'ifield index is larger than the number of '
        write(6,*) 'fields set to 2 '
        call exitt()
      endif
      RETURN
      END
c-----------------------------------------------------------------------

How do avoid this error? I changed my initial condition for temp from 0.0 to 1.0 but that didn't help.

Thank you

Nilay

**Comment by yslan** [comment]
Hi Nilay,

That usually happens if your BC or even vp is not set properly.
It might help if you set verbose = true under the [GENERAL] session and paste the residual and timestep as well.

Note that, the userbc inside the usr file is never called.
You should implement the boundary condition inside the oudf.
The reused BC value uin, vin and win need to be revisited as well.

Yu-Hsiang
--

**Reply by nkulkar2** [reply]
Hi Yu-Hsiang,

I applied the userbc and useric for a pipe flow case and it works with a recirculating flow condition and I get the flow profile as expected. And I tried adding verbose = true but the error persists. 

Thank you

Nilay
